src/Symfony/Component/Translation/Bridge/Loco/LocoProvider.php

```diff
@@ -75,8 +75,15 @@ public function write(TranslatorBagInterface $translatorBag): void
             }
 
             foreach ($catalogue->all() as $domain => $messages) {
-                $ids = $this->getAssetsIds($domain);
-                $this->translateAssets(array_combine($ids, array_values($messages)), $locale);
+                $keysIdsMap = [];
+
+                foreach ($this->getAssetsIds($domain) as $id) {
+                    $keysIdsMap[$this->retrieveKeyFromId($id, $domain)] = $id;
+                }
+
+                $ids = array_intersect_key($keysIdsMap, $messages);
+
+                $this->translateAssets(array_combine(array_values($ids), array_values($messages)), $locale);
```

Comment:
what if some keys don't have an asset id ? This would fail in `array_combine` AFAICT due to the different number of items.

Reply:
The case could not happen, Loco generate an asset for every asset.

Reply:
As we just create assets for new keys, they mandatory have an asset id